### DISCUSSION TITLE
Setting a feild to default when nothing is entered

### DISCUSSION THREAD

**Discussion by Argusham** [pocketbase/pocketbase#3818] [Q&A]
I have a collection called media which has a field called date that takes in a date and is required. if a user does not enter a date into the field  the field should default to the current date and time how can I achieve this

> 
	app.OnModelBeforeCreate("Media").Add(func(e *core.ModelEvent) error {
		rec := e.Model.(*models.Record)
		// Get the value of the date field
		Date := rec.GetTime("Date")
		// Check if the date is  
		if Date.IsZero() {

			// rec.Set("Date", time.Now())
			// Set the date to the current time
			rec.SetDataValue("date", time.Now())
		}
		
		return nil
	})

**Comment by ganigeorgiev** [comment]
The request hooks at the moment fire AFTER the request data load and validations, meaning that if you want the above hook to take effect you'll have to remove the "Nonempty" validation for the field (_this hooks behavior most likely will change in the future as several users complained about it_).

Alternatively, you can keep the "Nonempty" field check and instead of the hook submit the default date with your client-side request.
